crates/turbopack-ecmascript/src/chunk/source_map.rs

```diff
@@ -50,105 +50,29 @@ impl Asset for EcmascriptChunkSourceMapAsset {
-}
-
-#[turbo_tasks::value_impl]
-impl EcmascriptChunkEntrySourceMapAssetVc {
-    #[turbo_tasks::function]
-    pub fn new(chunk_path: FileSystemPathVc, chunk_item: EcmascriptChunkItemVc) -> Self {
-        EcmascriptChunkEntrySourceMapAsset {
-            chunk_path,
-            chunk_item,
-        }
-        .cell()
-    }
-}
-
-#[turbo_tasks::value_impl]
-impl Asset for EcmascriptChunkEntrySourceMapAsset {
-    #[turbo_tasks::function]
-    async fn path(&self) -> Result<FileSystemPathVc> {
-        // NOTE(alexkirsz) We used to asset's hash in the path, but this caused
-        // `all_assets_map` to be recomputed on every change.
-        Ok(self.chunk_path.append(&format!(
-            ".{}.map",
-            self.chunk_item.id().await?.to_truncated_hash()
-        )))
-    }
-
-    #[turbo_tasks::function]
-    async fn content(&self) -> Result<AssetContentVc> {
-        let sm = EcmascriptChunkContentEntryVc::new(self.chunk_item)
-            .await?
-            .code_vc
-            .generate_source_map()
-            .to_rope()
-            .await?;
-        Ok(File::from(sm).into())
-    }
-
-    #[turbo_tasks::function]
-    fn references(&self) -> AssetReferencesVc {
-        AssetReferencesVc::empty()
-    }
-}
-
 /// A reference to a [`EcmascriptChunkSourceMapAsset`], used to inform the dev
 /// server/build system of the presence of the source map
 #[turbo_tasks::value]
 pub struct EcmascriptChunkSourceMapAssetReference {
     chunk: EcmascriptChunkVc,
-    hot_module_replacement: bool,
 }
 
 #[turbo_tasks::value_impl]
 impl EcmascriptChunkSourceMapAssetReferenceVc {
     #[turbo_tasks::function]
-    pub fn new(chunk: EcmascriptChunkVc, hot_module_replacement: bool) -> Self {
-        EcmascriptChunkSourceMapAssetReference {
-            chunk,
-            hot_module_replacement,
-        }
-        .cell()
+    pub fn new(chunk: EcmascriptChunkVc) -> Self {
+        EcmascriptChunkSourceMapAssetReference { chunk }.cell()
     }
 }
 
-#[turbo_tasks::value(serialization = "auto_for_input")]
-#[derive(Debug, PartialOrd, Ord, Hash)]
-struct EcmascriptChunkEntrySourceMapAssetCellKey(EcmascriptChunkItemVc);
-
 #[turbo_tasks::value_impl]
 impl AssetReference for EcmascriptChunkSourceMapAssetReference {
     #[turbo_tasks::function]
     async fn resolve_reference(&self) -> Result<ResolveResultVc> {
-        let path = self.chunk.path();
-        let mut source_maps = Vec::new();
-        source_maps.push(
-            EcmascriptChunkSourceMapAsset { chunk: self.chunk }
-                .cell()
-                .into(),
-        );
-        if self.hot_module_replacement {
-            // Expose a SourceMap for each chunk item when HMR is enabled
-            for chunk_items_chunk in self
-                .chunk
-                .chunk_content_result()
-                .await?
-                .chunk_items
-                .await?
-                .iter()
-            {
-                for item in chunk_items_chunk.await?.iter() {
```

Comment:
This will improve HMR performance 🎉

Reply:
I suspect it'll also improve perf when loading the chunk's source map, too.